### DISCUSSION TITLE
Feature Request: Allow configuration of cache dir

### DISCUSSION THREAD

**Discussion by marrobHD** [vercel/next.js#58864] [Ideas]
### Goals

1. Ability to use NextJS cache in **immutable NixOS builds**
2. Ability to use NextJS cache in **Google App Engine Environment** & **AWS Lambda**
3. **Simplify CI** 
3. ability to alter the location of the .next/cache (ex. env var nextjs_cache)

### Non-Goals

1.
2.
4.

### Background

I saw that this [issue](https://github.com/vercel/next.js/issues/10111) got locked.
There definitely is demand for this feature. Due those locks there is no way to vote anymore.
I built a nix NextJS project package, which is read-only to make it immutable. That means there is no way to create a symlink.

There is a env for npm (npm_config_cache), which allows to configure the cache path. Something like that would be great for NextJS as well.

Due the missing implementation of a cache-dir-env in NextJS, I have to set "unoptimized: true", which significantly increases page loadtimes.

### Proposal

This feature could be implemented as nextjs config option & or as environment variable (ex. nextjs_cache)

**Comment by b2cc** [comment]
Running into other issues regarding this as well, e.g. here: https://github.com/lukevella/rallly/issues/1090

**Comment by alist** [comment]
Very interested in this due to my deployment in App Engine. Kinda silly that this doesn't exist.

**Comment by nicolo-tito** [comment]
This affects also deployment to Azure as a Node web app, since by default the app is mounted as a package in /home/site/wwwroot https://learn.microsoft.com/en-us/azure/azure-functions/run-functions-from-deployment-package, so the Next app can't access the cache folder. 
```
2024-07-03T15:26:39.081055556Z: [ERROR]  Failed to update tags manifest. [Error: ENOENT: no such file or directory, mkdir '/home/site/wwwroot/.next/cache'] {
2024-07-03T15:26:39.081080457Z: [ERROR]    errno: -2,
2024-07-03T15:26:39.081134357Z: [ERROR]    code: 'ENOENT',
2024-07-03T15:26:39.081142557Z: [ERROR]    syscall: 'mkdir',
2024-07-03T15:26:39.081147757Z: [ERROR]    path: '/home/site/wwwroot/.next/cache'
2024-07-03T15:26:39.081153057Z: [ERROR]  }
```
Since /wwwroot is a mounted read-only folder, this can't be fixed by setting a symbolic link to a writable folder. Being able to configure the cache dir would allow to solve this. 
Right now the only way to fix this is deploy Next as a docker container I guess, which is way more cumbersome then just deploying the standalone build to a default Azure Node web app resource.

**Comment by pealmeid** [comment]
Also interested in this, our use case is the same as @nicolo-tito.

**Comment by andreasnett** [comment]
I have the exact same problem not allowing update for cache manifest. Did you find a workaround @nicolo-tito ?

**Reply by nicolo-tito** [reply]
I'm afraid there is no workaround for this. Because of this we just decided to deploy the application as a container web app (which does remove the problem, but as I mentioned is quite an effort, while @marrobHD config option proposal would make this very easy to solve).

**Comment by bcmedeiros** [comment]
it's just ridiculous this doesn't exist.
